metrics/micrometer/src/main/java/com/datastax/oss/driver/metrics/micrometer/MicrometerDriverContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.metrics.micrometer;
+
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.metrics.MetricsFactory;
+import com.datastax.oss.driver.api.core.session.ProgrammaticArguments;
+import com.datastax.oss.driver.internal.core.context.DefaultDriverContext;
+import io.micrometer.core.instrument.MeterRegistry;
+
+/** Implementation of {@link DriverContext} that provides for a Micrometer {@link MeterRegistry}. */
```

Comment:
The `DefaultDriverContext` is using `Dropwizard` metrics. 
Maybe to make it consistent we should create a `DropwizardDriverContext extends DefaultDriverContext ` to provide our users with a consistent API for using metrics:
- `DropwizardDriverContext`
- `MicrometerDriverContext`
- `MicroProfileDriverContext`
If not, then we should at least document that.